### ISSUE TITLE
feat(linter): eslint/no-unexpected-multiline

### ISSUE THREAD

**Issue by DonIsaac** [web-infra-dev/oxc#620]
Implement ESLint's [`no-unexpected-multiline`](https://eslint.org/docs/latest/rules/no-unexpected-multiline#rule-details) rule.

To get started, run
```sh
just new-rule no-unexpected-multiline
```

**Comment by EYHN** [comment]
I just started looking at this repository and I tried to start with this issue, then I did some research and I realized that it is hard for me to implement this rule in a simple way.
Imitating eslint's implementation, I need to find the corresponding Statement from AstNode in Program, and since AstNode doesn't have an index stored link to the Statement, I do this step by doing a binary search .
Then I can find the previous Statement of the current Statement to determine if they are on the same line, but i can't find the line number of the Statement, so I can only locate the line number by traversing the line breaks in the source code.
Both of these binary search and traversing the source code are slower operations, and if the performance drop is unacceptable, to implement this rule I may have to store more information such as the line number, AstNode to Statement Index.
